### ISSUE TITLE
Add tests with learning rate schedule to weight_decay_optimizers_test.py

### ISSUE THREAD

**Issue by sfleisch** [tensorflow/addons#866]
Currently tests in weight_decay_optimizers_test.py only test constant learning rate and weight decay parameters.  It would be useful to also have test cases that use tf.keras.optimizers.learning_rate_schedule for which support apparently was added with the current tfa 0.7.0.

**Comment by WindQAQ** [comment]
Sure, test coverage should be as large as possible as scheduled weight decay is requested more than one time. I am going to close this issue because I believe while addressing #865, this should be done too. Feel free to re-open it if you have further questions. Thank you!
